features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/metric/SessionEndedMetric.kt

```diff
@@ -74,14 +84,45 @@ internal class SessionEndedMetric(
         } ?: 0L
     }
 
-    private fun resolveRseAttributes(): Map<String, Any?> {
+    private fun resolveRseAttributes(ntpOffsetAtEnd: Long): Map<String, Any?> {
         return mapOf(
             PROCESS_TYPE_KEY to PROCESS_TYPE_VALUE,
             PRECONDITION_KEY to startReason.asString,
             DURATION_KEY to calculateDuration(),
             WAS_STOPPED_KEY to wasStopped,
             VIEW_COUNTS_KEY to resolveViewCountsAttributes(),
-            SDK_ERRORS_COUNT_KEY to resolveSDKErrorsCountAttributes()
+            SDK_ERRORS_COUNT_KEY to resolveSDKErrorsCountAttributes(),
+            NO_VIEW_EVENTS_COUNT_KEY to resolveNoViewCountsAttributes(),
+            HAS_BACKGROUND_EVENTS_TRACKING_ENABLED_KEY to hasTrackBackgroundEventsEnabled,
+            NTP_OFFSET_KEY to resolveNtpOffsetAttributes(ntpOffsetAtEnd)
+        )
+    }
+
+    private fun resolveNoViewCountsAttributes(): Map<String, Any?> {
+        return mapOf(
+            NO_VIEW_EVENTS_COUNT_ACTIONS_KEY to (
+                missedEventCountByType[MissedEventType.ACTION]
+                    ?: 0
+                ),
+            NO_VIEW_EVENTS_COUNT_RESOURCES_KEY to (
+                missedEventCountByType[MissedEventType.RESOURCE]
+                    ?: 0
+                ),
+            NO_VIEW_EVENTS_COUNT_ERRORS_KEY to (
+                missedEventCountByType[MissedEventType.ERROR]
+                    ?: 0
+                ),
+            NO_VIEW_EVENTS_COUNT_LONG_TASKS_KEY to (
+                missedEventCountByType[MissedEventType.LONG_TASK]
+                    ?: 0
+                )
+        )
+    }
+
+    private fun resolveNtpOffsetAttributes(ntpOffsetAtEnd: Long): Map<String, Any?> {
```

Comment:
same as above, no need to widen the value type:

```suggestion
    private fun resolveNtpOffsetAttributes(ntpOffsetAtEnd: Long): Map<String, Long> {
```

Reply:
fixed